app/frontend/wallet/shelley-wallet.ts

```diff
@@ -71,6 +72,18 @@ const ShelleyWallet = ({config, cryptoProvider}: WalletParams) => {
     )
   }
 
+  function getTokensMetadata(
+    accountInfo: Array<AccountInfo>
+  ): Promise<{[subject: string]: RegisteredTokenMetadata}> {
+    return TokenRegistryApi().getTokensMetadata([
```

Comment:
this looks a bit artificial and makes one question why TokenRegistryApi isn't either a proper class with configurable parameters like the server URL, or a simple helper function that would be directly called.

I'd consider making a TokenRegistry (API being a redundant suffix IMHO) class that would take in its constructor the server URL (similarly as BlockchainExplorer does, but it isn't a class - pre-TS legacy) and would offer the getTokenMetada method. parseTokenMetadata could remain a standalone helper function used within the class which we would export from the module just for the sake of unit-testing

Reply:
Refactored